src/Compilers/CSharp/Portable/Lowering/MethodToClassRewriter.cs

```diff
@@ -632,10 +633,10 @@ public override BoundNode VisitReadOnlySpanFromArray(BoundReadOnlySpanFromArray
             return node.Update(operand, method, type);
         }
 
-        private static bool BaseReferenceInReceiverWasRewritten(BoundExpression originalReceiver, BoundExpression rewrittenReceiver)
+        private static bool BaseReferenceInReceiverWasRewritten([NotNullWhen(true)] BoundExpression? originalReceiver, [NotNullWhen(true)] BoundExpression? rewrittenReceiver)
         {
-            return originalReceiver != null && originalReceiver.Kind == BoundKind.BaseReference &&
-                   rewrittenReceiver != null && rewrittenReceiver.Kind != BoundKind.BaseReference;
+            return originalReceiver is { Kind: BoundKind.BaseReference } &&
+                   rewrittenReceiver is { Kind: BoundKind.BaseReference };
```

Comment:
>rewrittenReceiver is { Kind: BoundKind.BaseReference }; [](http://example.com/codeflow?start=19&length=55)

Consider not bundling refactorings like this with product changes in the same PR. This Also doesn't look like a change related to enabling nullable annotations. It is a good thing we have tests that cought this bug.